compound_gene/compound_binds_gene/dataset_statistics/dataset_statistics.py

```diff
@@ -0,0 +1,527 @@
+        CompoundGene.id.in_(
+            total_candidates_df
+            .query("split==7")
+            .sample(10000, random_state=100)
+            .candidate_id
+            .tolist()
+        )
+    )
+    .all()
+)
+dev_df = make_sentence_df(dev_candidates)
+dev_df.head(2)
+
+
+# In[18]:
+
+
+test_candidates = (
+    session
+    .query(CompoundGene)
+    .filter(
+        CompoundGene.id.in_(
+            total_candidates_df
+            .query("split==8")
+            .sample(10000, random_state=100)
+            .candidate_id
+            .tolist()
+        )
+    )
+    .all()
+)
+test_df = make_sentence_df(test_candidates)
+test_df.head(2)
+
+
+# In[ ]:
+
+
+write_candidates_to_excel(dev_df, "data/sentence_labels_dev.xlsx")
+write_candidates_to_excel(test_df, "data/sentence_labels_test.xlsx")
+
+
+# ## Distribution of Sentence Length
+
+# In[19]:
+
+
+sns.distplot(total_candidates_df["sen_length"], rug=False)
+
+
+# In[20]:
+
+
+total_candidates_df["sen_length"].describe().astype(int)
+
+
+# Something seems fishy about this distribution. The number of words (tokens) for a given sentence is in the thousands range. Intuitively, that doesn't make sense, since the average number of words for a given sentence is 42. Possible reason for this abnormality is a parsing error. Lets take a look at this 834 word sentence.
+
+# In[21]:
+
+
+total_candidates_df.query("sen_length==834").iloc[0]["text"]
+
+
+# The above sentence shows that a long list of drugs that a issue will cover. This isn't helpful in our case, because we want to look at sentences that contain compounds and diseases; therefore, we should remove sentences of this nature by defining a cutoff score of 83 or less words.
+
+# In[22]:
+
+
+sns.distplot(total_candidates_df.query("sen_length < 83+1")["sen_length"], rug=False)
+
+
+# In[23]:
+
+
+total_candidates_df.query("sen_length < 83+1")["sen_length"].describe().astype(int)
+
+
+# This distribution looks a bit more reasonable compared to the above distribution. After filtering out the outliers, we still have a pleathora of sentences on the order of 2.2 million.
+
+# In[24]:
+
+
+before_filter = set([tuple(line) for line in total_candidates_df[["drugbank_id", "entrez_gene_id"]].values])
+after_filter = set([tuple(line) for line in total_candidates_df.query("sen_length < 83+1")[["drugbank_id", "entrez_gene_id"]].values])
+print(
+    "Total number of unique candidates before filter: {}".format(
+        total_candidates_df[["drugbank_id", "entrez_gene_id"]].drop_duplicates().shape[0]
+    )
+)
+print(
+    "Total number of unique candidates after filter: {}".format(
+        total_candidates_df.query("sen_length < 83+1")[["drugbank_id", "entrez_gene_id"]].drop_duplicates().shape[0]
+    )
+)
+print("Total number of unique candidates being thrown out: {}".format(len(before_filter.difference(after_filter))))
+
+
+# In[25]:
+
+
+filtered_total_candidates_df = total_candidates_df.query("sen_length < 83+1")
+
+
+# In[26]:
+
+
+venn2(
```

Comment:
Is it surprising how large the discrepancy is?

Reply:
As discussed in person. This large discrepancy is surprising. Possible reasons for this could be:  
1. A significant amount of time has passed (~ 2 years) and the large amount of new publications is causing the red circle to be so large.
2. The databases used to capture the compound binds gene relationships don't cover as much of these Pubmed abstracts as we would hope.
3. Majority of the sentences that belong in the red circle are just spurious mentions, which means these types of sentences are not informative in terms of asserting there is a relation.